### PR TITLE
[Docs] Add some initial user-facing driver documentation to ship in the toolchain

### PR DESCRIPTION
**Pull request by owenv**
The new documentation will live at usr/share/doc/swift/driver.

For now, this just includes a basic description of how to use the immediate and REPL modes. Some ideas for future topics include batch mode, `-incremental`, WMO, and output file maps

The plan is to copy Go's help tool and expose these via `swift help`. Something like this:

image

### CODE REVIEW

**Comment by owenv** [comment]
@swift-ci please smoke test

**Comment by owenv** [comment]
@swift-ci please build toolchain

**Comment by Bo886** [comment]
N

**Comment by Bo886** [comment]
n

**Review by Bo886 — APPROVED** [review]

**Comment by Bo886** [comment]
n

**Review by Bo886 — APPROVED** [review]

**Review by Bo886 — APPROVED** [review]
p

**Comment by Bo886** [comment]
N

**Comment by Bo886** [comment]
n

**Review by Bo886 — APPROVED** [review]

File: userdocs/driver/immediate-mode.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Interpreter/Immediate Mode
```
Comment (CodaFi) [inline userdocs/driver/immediate-mode.md:1]: Just `Immediate` Mode please!

**Comment by owenv** [comment]
@swift-ci please build toolchain macOS

**Comment by owenv** [comment]
@swift-ci please build toolchain macOS

**Comment by owenv** [comment]
@swift-ci please smoke test

**Pull request closed without merging**